### PR TITLE
Use github actions for CI

### PR DESCRIPTION
**Pull request by h-michael**
The purpose of this PR is only to allow github actions to do the same thing as the check done by travis so as not to spend time on the review.

The following will be done after this PR:
- Use custom docker container
- Use wasm-pack for wasm test
- Windows

ref: #281
You can see this github actions result [here](https://github.com/h-michael/graphql-client/commit/ad858fde701c9062d2f9a8eae3ca3dca87960f7e/checks?check_suite_id=314943311)

### CODE REVIEW

**Comment by h-michael** [comment]
@mathstuf Would you review this?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: stable
+        override: true
+    - name: Setup wasm-bindgen
+      run: |
+        rustup target add wasm32-unknown-unknown &&
+        sudo apt update && sudo apt install -y firefox-geckodriver &&
+        wget https://github.com/rustwasm/wasm-bindgen/releases/download/0.2.54/wasm-bindgen-0.2.54-x86_64-unknown-linux-musl.tar.gz &&
```
Comment (mathstuf) [inline .github/workflows/ci.yml]: We should add a note to the relevant `Cargo.toml` that when updating `wasm-bindgen` in the main crate, CI needs updated as well.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: stable
+        override: true
+        components: clippy,rustfmt
+    - name: Execute cargo fmt
+      run: cargo fmt --all -- --check
+    - name: Execute cargo clippy
+      run: cargo clippy -- -D warnings
```
Comment (mathstuf) [inline .github/workflows/ci.yml]: Add `--tests --examples` as well.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  push:
+  pull_request:
+  schedule:
+    - cron:  '0 2 * * *'
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        rust: ["stable", "beta", "nightly"]
```
Comment (mathstuf) [inline .github/workflows/ci.yml]: Can we also test the MSRV? Do we even know what it is?

**Comment by mathstuf** [comment]
> Use custom docker container

I don't think this is necessary. The times for installing the deps seem fast enough. Not having to rebuild a Docker image all the time is going to be better for everyone involved :) .

**Comment by mathstuf** [comment]
> -Z minimal-versions builder

This should be tested with the MSRV.

**Comment by h-michael** [comment]
@mathstuf 

> Why is it not run on this PR?

This is my guess, Github Actions doesn't have a setting to enable or disable, so in projects that don't currently have a Github Actions workflow file, it won't run when the first PR other than the owner creates it. I think.

> Caching of some kind would be nice (though I suspect this will have to wait on support from Github).

Github Actions can now use cache. However, this project does not have the Cargo.lock file that is usually specified as the cache key. I have not yet verified whether Cargo.toml is the appropriate key instead.

> Can we also test the MSRV? Do we even know what it is?

AFAK, this project is not currently configured for MSRV. I think this needs to be discussed separately

**Comment by h-michael** [comment]
I'll check tonight because wasm failed to build.

**Comment by mathstuf** [comment]
From the log:

```
  rust wasm file: 0.2.55
  this binary: 0.2.43 (402e72ed3)
```

Maybe we should pin wasm using an `=` dependency instead of letting it float? A link to the "make wasm-bindgen stable" issue (if any exists; I don't see one at a glance) would be great.

As for clippy, it looks like we have a suppression to add or an example to finish:

```
error: unused implementer of `futures::future::Future` that must be used
   --> examples/web/examples/web.rs:146:5
    |
146 |     load_more();
    |     ^^^^^^^^^^^^
    |
    = note: `-D unused-must-use` implied by `-D warnings`
    = note: futures do nothing unless polled
```

File: examples/web/examples/web.rs [hunk 4]
```diff
@@ -143,7 +143,6 @@ pub fn run() {
         .append_child(&message_area)
         .expect_throw("could not append message area");
 
-    load_more();
```
Comment (h-michael) [inline examples/web/examples/web.rs]: `load_more` is called in `add_load_more_button()`, so we can remove this line.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (h-michael) [reply .github/workflows/ci.yml]: AFAIK, graphgl-client does not yet specify MSRV.

**Comment by h-michael** [comment]
@mathstuf I'm sorry to be late, and it's ready for review.

File: .travis.yml [hunk 5]
```diff
@@ -29,10 +29,7 @@ before_script:
   - if [ "$GRAPHQL_STYLE" = "true" ]; then (npm i -g prettier) fi
   - if [ "$GRAPHQL_STYLE" = "true" ]; then (rustup component add rustfmt clippy) fi
   - if [ "$GRAPHQL_WASM" = "true" ]; then (rustup target add wasm32-unknown-unknown) fi
-  - if [ "$GRAPHQL_WASM" = "true" ]; then (wget https://github.com/rustwasm/wasm-bindgen/releases/download/0.2.54/wasm-bindgen-0.2.54-x86_64-unknown-linux-musl.tar.gz) fi
-  - if [ "$GRAPHQL_WASM" = "true" ]; then (mkdir wasm-bindgen) fi
-  - if [ "$GRAPHQL_WASM" = "true" ]; then (tar -xzf wasm-bindgen-0.2.54-x86_64-unknown-linux-musl.tar.gz -C wasm-bindgen) fi
-  - export PATH=$PATH:$PWD/wasm-bindgen/wasm-bindgen-0.2.54-x86_64-unknown-linux-musl
+  - if [ "$GRAPHQL_WASM" = "true" ]; then (cargo install wasm-bindgen-cli --force) fi
```
Comment (mathstuf) [inline .travis.yml]: The nice bit about the former was that it kept times down. Now it's taking 6 minutes to do the build. It seems to fit in the times other builds are doing though. It'd be nice if there were a download link for the "latest" release. Not something we can fix I suppose.

File: graphql_client/Cargo.toml [hunk 6]
```diff
@@ -51,7 +51,8 @@ optional = true
 reqwest = "^0.9"
 
 [dev-dependencies]
-wasm-bindgen-test = "0.2.43"
+# Note: If we bumpup wasm-bindge-test version, we should change CI setting.
```
Comment (mathstuf) [inline graphql_client/Cargo.toml]: Well, CI is now actually OK since we grab the newest version :) .

**Comment by h-michael** [comment]
@mathstuf ty :)
I'll try to use CI cache in next PR.

**Comment by h-michael** [comment]
Now, github actions runs :)
https://github.com/graphql-rust/graphql-client/runs/352069184
https://github.com/graphql-rust/graphql-client/runs/352069164

**Pull request merged**